cmd/api/src/api/v2/auth/sso.go

```diff
@@ -122,3 +129,33 @@ func (s ManagementResource) ListAuthProviders(response http.ResponseWriter, requ
 		}
 	}
 }
+
+// DeleteSSOProviderResponse represents the response returned to the user from DeleteSSOProvider
+type DeleteSSOProviderResponse struct {
+	AffectedUsers model.Users `json:"affected_users"`
+}
+
+// DeleteSSOProvider deletes a sso_provider with the matching id
+func (s ManagementResource) DeleteSSOProvider(response http.ResponseWriter, request *http.Request) {
+	var (
+		rawSSOProviderID = mux.Vars(request)[api.URIPathVariableSSOProviderID]
+		requestContext   = ctx.FromRequest(request)
+	)
+
+	// Convert the incoming string url param to an int
+	if ssoProviderID, err := strconv.Atoi(rawSSOProviderID); err != nil {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, err.Error(), request), response)
+	} else if user, isUser := auth.GetUserFromAuthCtx(requestContext.AuthCtx); isUser && user.SSOProviderID.Equal(null.Int32From(int32(ssoProviderID))) {
+		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusConflict, "user may not delete their own SSO auth provider", request), response)
+	} else if providerUsers, err := s.db.GetSSOProviderUsers(request.Context(), ssoProviderID); err != nil {
+		api.HandleDatabaseError(request, response, err)
```

Comment:
I think this also needs the NotFound handling treatment as below.

Reply:
This should be fine since a user may want to delete an SSO Provider that has not been assigned to any users yet.

Reply:
Handled offline but the snag was in the db method preload, not here with users! Resolving